Fix customfields sorting

danielwolf1 opened the pull request:
### 1. Why is this change necessary?
Custom fields where not sorted naturally by customFieldPosition.
Before: 1,10,9
With this change: 1,9,10

### 2. What does this change do, exactly?
It sets the sort naturally flag on the custom fields.

### 3. Describe each step to reproduce the issue or behaviour.
Add product custom fields with customFieldPosition 1, 10 and 9

### 4. Please link to the relevant issues (if any).


### 5. Checklist

- [ ] I have written tests and verified that they fail without my change
- [x] I have squashed any insignificant commits
- [ ] I have written or adjusted the documentation according to my changes
- [x] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

janbuecker commented:
Hi & thanks for your PR.

Currently, it is possible to use base store id's like this:

```
            $form->setElement('combo', 'customer_groups', 
                array(
                    'label' => 'Kundengruppen',
                    'store' => 'base.Payment',
                    'multiSelect' => true,
                    'scope' => Shopware\Models\Config\Element::SCOPE_SHOP,
                    'value' => '',
                )
            );
```

svenmuennich commented:
Yes, but the current implementation has side effects, e.g. when random plugins add filters to a `base` store, since these stores are singletons. Please see also [my comment](https://github.com/shopware/shopware/pull/525#issue-151290059) in the already merged PR #525

svenmuennich commented:
This PR is actually more a fix, since using the new store capabilities added in 3fc607dd19ba4c7092a4b42f2aa632c153a7f67d causes an error in ExtJs when opening the plugin config via the settings module: `Uncaught TypeError: Cannot read property 'on' of undefined`

I forgot to dass this to the description.

janbuecker commented:
I do not encounter any problems neither in the plugin config nor in the base configuration. It is necessary to define the store with this kind of id `'store' => 'base.Payment'` and not `'store' => 'Shopware.apps.Base.store.Payment'`.

svenmuennich commented:
Yes, it still works using e.g. `base.Payment`. However, in that case the store is initialised using `getStore()`, which might be an instance of a e.g. the `payment` store, which some plugin as added filters to. We encountered this issue several times in our plugins, where e.g. not all payment methods would show up for selection in the config. Hence we opened PR #525 to fix this, by ensuring that a new instance of the store is created, when passing the full name of a store (e.g. `Shopware.apps.Base.store.Payment`).

The point of this PR is to achieve compatibility of the `Shopware.apps.Config.view.element.Select` for the new store loading. You can reproduce this error I mentioned [here](https://github.com/shopware/shopware/pull/600#issuecomment-227091327) by replacing `base.Payment` with `Shopware.apps.Base.store.Payment` in on of your config elements. You will still be able to select payment methods when opening the plugin config via the plugin manager, but get an error when trying to do the same via the settings module.

janbuecker commented:
I've gone a step further and refactored the whole component since it's nearly an 1:1 copy of the base combo box. 

Merged with c301e70a2eacd1b5c99fc5fa6fb1ca714906740e

svenmuennich commented:
Great, thank you 👍

danielwolf1 committed:
Fix customField sorting
Make sure customFields are sorted naturally by ascdening numbers:
1,9,10 and not 1,10,9

danielwolf1 committed:
update doc

jleifeld commented:
Hey @danielwolf1 ,

thanks for your contribution 👍 

This looks like a quickPick, therefore I will merge your PR soon 🙂

mitelg commented:
was merged with https://github.com/shopware/platform/commit/1f6287397a883b78c26c8aa18a4042b99a469550
thanks for your contribution @danielwolf1 :+1: :tada: :blue_heart:
